library/std/src/sys/pal/unix/stack_overflow.rs

```diff
@@ -25,15 +25,35 @@ impl Drop for Handler {
     }
 }
 
-#[cfg(any(
-    target_os = "linux",
-    target_os = "freebsd",
-    target_os = "hurd",
-    target_os = "macos",
-    target_os = "netbsd",
-    target_os = "openbsd",
-    target_os = "solaris",
-    target_os = "illumos",
+#[cfg(all(
+    not(miri),
+    any(
+        target_os = "linux",
+        target_os = "freebsd",
+        target_os = "hurd",
+        target_os = "macos",
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "solaris",
+        target_os = "illumos",
+    ),
+))]
+mod thread_info;
+
+// miri doesn't model signals, and this code has some synchronization properties
+// that we don't want to expose to user code.
```

Comment:
```suggestion
// that we don't want to expose to user code. Also, Miri fundamentally cannot even have a stack overflow.
```

Reply:
How could Miri have stack overflows? The stack is a `Vec<Frame>`. We can have a `push` fail due to OOM, but that will just abort the interpreter aprubtly in a well-defined way.